File: bouncing-balls.cpp
#include <iostream>
using namespace std;

int main() {
    double startingHeight;
    cin >> startingHeight;
    
    double firstBounceHeight;
    cin >> firstBounceHeight;
    
    double bouncinessIndex = firstBounceHeight / startingHeight;
    
    int numBounces;
    cin >> numBounces;
    
    double totalDistance = 2.0 * (1 - pow(bouncinessIndex, numBounces)) * startingHeight;
    
    cout.setprecision(10);
    cout << fixed << totalDistance << endl;

    return 0;
}

File: bouncing-balls.py
```
def bouncing_balls():
    start_height = float(input())
    first_bounce_height = float(input())
    num_bounces = int(input())

    bounciness_index = first_bounce_height / start_height
    total_distance = 0.0

    for _ in range(num_bounces):
        height = (2 * start_height + 2 * first_bounce_height) ** (1 / 2)
        total_distance += height
        start_height, first_bounce_height = (
            first_bounce_height,
            start_height * bounciness_index,
        )

    print(bounciness_index)
    print(total_distance)


bouncing_balls()